src/iidm/Substation.cpp

```diff
@@ -124,6 +128,47 @@ VoltageLevelAdder Substation::newVoltageLevel() {
     return VoltageLevelAdder(*this);
 }
 
+void Substation::remove() {
+    Substations::checkRemovability(*this);
+
+    std::vector<std::reference_wrapper<VoltageLevel>> voltageLevels;
+    for (VoltageLevel& vl : m_voltageLevels) {
+        voltageLevels.emplace_back(vl);
+    }
+
+    for (VoltageLevel& vl : voltageLevels | boost::adaptors::reversed) {
```

Comment:
I think this is a bit complicated: as you don't remove items from the vector itself, I think it's possible to iterate over `m_voltageLevels` and then clear the vector.

Reply:
Agree for this one => updated